components/prompt-form.tsx

```diff
@@ -45,16 +47,87 @@ export function PromptForm({
   const messages = useSelector((state: any) => state.chat.messages)
   const threadId = useSelector((state: any) => state.chat.threadId)
   const pathname = usePathname()
+  const [selectedFiles, setSelectedFiles] = React.useState<FileData[]>([])
+  const [isAssistantRunning, setIsAssistantRunning] =
+    React.useState<boolean>(false)
 
-  async function submitUserMessage(messageId: string, value: string) {
-    const createdAt = new Date()
-    const firstMessageContent = value as string
-    const title = firstMessageContent.substring(0, 100)
+  const submitUserMessage = async (
+    messageId: string,
+    value: string,
+    files: FileData[]
+  ) => {
+    const { currentThreadId, chat } = await getCurrentChat(
+      messageId,
+      value,
+      files
+    )
 
+    await openai.beta.threads.messages.create(currentThreadId, {
+      role: 'user',
+      content: value
+    })
+
+    const stream = await openai.beta.threads.runs.stream(currentThreadId, {
+      assistant_id: openAIAssistantId || '',
+      stream: true
+    })
+
+    let assistantResponse = ''
+    const newAssistantChatId = nanoid()
+
+    for await (const message of stream) {
+      if (
+        message.event === 'thread.message.delta' &&
+        message.data.delta.content
+      ) {
+        const text = (message.data.delta.content[0] as any).text.value
+          ? (message.data.delta.content[0] as any).text.value
+          : ''
+        assistantResponse += text
+
+        dispatch(
+          addMessage({
+            id: newAssistantChatId,
+            message: assistantResponse,
+            role: 'assistant'
+          })
+        )
+      }
+    }
+
+    const assistantMessage = {
+      id: newAssistantChatId,
+      message: assistantResponse,
+      role: 'assistant'
```

Comment:
Similar comment to https://github.com/Conrad-Labs/nextjs-chatbot-ui-template/pull/3/files#r1860588476